treeherder/model/models.py

```diff
@@ -136,6 +137,55 @@ class Meta:
     def __str__(self):
         return "{0}".format(self.id)
 
+    @classmethod
+    def search(cls, search_term):
+        max_size = 50
+        # 90 days ago
+        time_limit = datetime.datetime.now() - datetime.timedelta(days=90)
+        # Wrap search term so it is used as a phrase in the full-text search.
+        search_term_fulltext = search_term.join('""')
+        # Substitute escape and wildcard characters, so the search term is used
+        # literally in the LIKE statement.
+        search_term_like = search_term.replace('=', '==').replace(
+            '%', '=%').replace('_', '=_')
+        recent = cls.objects.raw(
+            '''
+            SELECT id, summary, crash_signature, keywords, os, resolution,
+            MATCH (`summary`) AGAINST (%s IN BOOLEAN MODE) AS relevance
+            FROM bugscache
+            WHERE 1
+              AND resolution = ''
+              AND `summary` LIKE CONCAT ('%%%%', %s, '%%%%') ESCAPE '='
+              AND modified >= %s
+            ORDER BY relevance DESC
+            LIMIT 0,%s
+            ''', [search_term_fulltext, search_term_like, time_limit,
+                  max_size])
+
+        all_others = cls.objects.raw(
+            '''
+            SELECT id, summary, crash_signature, keywords, os, resolution,
+            MATCH (`summary`) AGAINST (%s IN BOOLEAN MODE) AS relevance
+            FROM bugscache
+            WHERE 1
+            AND `summary` LIKE CONCAT ('%%%%', %s, '%%%%') ESCAPE '='
+            AND (modified < %s OR resolution <> '')
+            ORDER BY relevance DESC
+            LIMIT 0,%s''', [search_term_fulltext, search_term_like, time_limit,
+                            max_size])
+
+        def _bug_dict(bug):
+            return {
+                'crash_signature': b.crash_signature,
```

Comment:
I know these are pre-existing, but this should either be `bug.crash_signature` (and so on), or else the parameter name for `_bug_dict()` should be named `b`.